### ISSUE TITLE
editor bug with variational / Optimize

### ISSUE THREAD

**Issue by dritchie** [probmods/ppaml2016#2]
If you run this code in a code box

``` javascript
var fitGaussian = function() {
  var mu = scalarParam(0, 1);
  var sigma = Math.exp(scalarParam(0, 1));
  factor(NaN);
};

Optimize(fitGaussian, {steps: 1000});
```

you'll get an error (as intended). Now, if you try to run any code in the same code box after this, you'll get the error

```
Address prefix mismatch.
```

Seems like there's something lingering around in the global coroutine from the previous execution that shouldn't be.

**Comment by null-a** [comment]
> Seems like there's something lingering around in the global coroutine from the previous execution that shouldn't be.

Yeah, I think this is caused by the same problem as #1. I'll take a look.

**Comment by null-a** [comment]
Closed by probmods/webppl#565.
